js-packages/@prestojs/ui-antd/src/getWidgetForField.ts

```diff
@@ -46,19 +46,20 @@ export default function getWidgetForField<FieldValue, T extends HTMLElement>(
     field: Field<FieldValue>
 ): FieldWidgetType<FieldValue, T> | [FieldWidgetType<FieldValue, T>, object] | null {
     // Couldn't work out what to type this as so field.constructor was accepted
-    const widget: FieldWidgetType<any, any> | null | undefined = field.choices
-        ? choicesMapping.get(field.constructor.name) || mapping.get(field.constructor.name)
-        : mapping.get(field.constructor.name);
+    const widget: FieldWidgetType<any, any> | null | undefined =
+        field.choices || field.asyncChoices
+            ? choicesMapping.get(field.constructor.name) || mapping.get(field.constructor.name)
+            : mapping.get(field.constructor.name);
```

Comment:
This is `(field.choices || field.asyncChoices) ? ... : `? -- it might be worth putting brackets around that to make it a bit clearer what the order of precedence is (or maybe it's just me not seeing enough of these JS patterns)